src/hooks/useSnackMessage.ts

```diff
@@ -23,81 +28,57 @@ export interface UseSnackMessageReturn {
     const intlRef = useIntlRef();
-    const { enqueueSnackbar } = useSnackbar();
+    const { enqueueSnackbar, closeSnackbar } = useSnackbar();
+
+    const enqueue = useCallback(
+        (snackInputs: SnackInputs, variant: BaseVariant) => {
+            const message = makeMessage(intlRef, snackInputs);
+            return enqueueSnackbar(message, {
+                ...snackInputs,
+                variant: variant,
+                style: { whiteSpace: 'pre-line' },
+            });
+        },
+        [enqueueSnackbar, intlRef]
+    );
 
     /*
         There is two kind of messages : the message itself (bottom of snackbar), and the header (top of snackbar).
         As inputs, you can give either a text message, or an ID with optional values (for translation with intl).
-            snackInputs: {
-                messageTxt,
-                messageId,
-                messageValues,
-                headerTxt,
-                headerId,
-                headerValues,
+          snackInputs: {
+              messageTxt,
+              messageId,
+              messageValues,
+              headerTxt,
+              headerId,
+              headerValues,
+              key?, // optional key to close the snackbar
+              persist
             }
    */
     const snackError = useCallback(
-        (snackInputs: SnackInputs) =>
-            makeSnackbar(snackInputs, intlRef, enqueueSnackbar, 'error', true),
-        [enqueueSnackbar, intlRef]
+        (snackInputs: SnackInputs) => enqueue(snackInputs, 'error'),
+        [enqueue]
     );
 
     /* see snackError */
     const snackWarning = useCallback(
-        (snackInputs: SnackInputs) =>
-            makeSnackbar(
-                snackInputs,
-                intlRef,
-                enqueueSnackbar,
-                'warning',
-                true
-            ),
-        [enqueueSnackbar, intlRef]
+        (snackInputs: SnackInputs) => enqueue(snackInputs, 'warning'),
+        [enqueue]
     );
 
     /* see snackError */
     const snackInfo = useCallback(
-        (snackInputs: SnackInputs) =>
-            makeSnackbar(snackInputs, intlRef, enqueueSnackbar, 'info', false),
-        [enqueueSnackbar, intlRef]
+        (snackInputs: SnackInputs) => enqueue(snackInputs, 'info'),
+        [enqueue]
     );
 
-    return { snackError, snackInfo, snackWarning };
-}
-
-function makeSnackbar(
-    snackInputs: SnackInputs,
-    intlRef: React.MutableRefObject<IntlShape>,
-    enqueueSnackbar: EnqueueSnackbar,
-    level: BaseVariant,
-    persistent: boolean
-) {
-    const message = checkAndTranslateIfNecessary(
-        intlRef,
-        snackInputs.messageTxt,
-        snackInputs.messageId,
-        snackInputs.messageValues
-    );
-    const header = checkAndTranslateIfNecessary(
-        intlRef,
-        snackInputs.headerTxt,
-        snackInputs.headerId,
-        snackInputs.headerValues
-    );
-    if (message !== null && header !== null) {
```

Comment:
I'm afraid we have a regression here, because previously if one is null we don't display the snackbar. We should keep it to be sure we have no important regression. Even if it is certainly not the main use case.

Reply:
done